src/Sylius/Behat/Page/Admin/Product/IndexPage.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Admin\Product;
+
+use Behat\Mink\Session;
+use Sylius\Behat\Page\SymfonyPage;
+use Sylius\Behat\TableManipulatorInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class IndexPage extends SymfonyPage implements IndexPageInterface
+{
+    /**
+     * @var TableManipulatorInterface
+     */
+    private $tableManipulator;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param TableManipulatorInterface $tableManipulator
+     */
+    public function __construct(
+        Session $session,
+        array $parameters,
+        RouterInterface $router,
+        TableManipulatorInterface $tableManipulator
+    ) {
+        parent::__construct($session, $parameters, $router);
+
+        $this->tableManipulator = $tableManipulator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isThereProduct(ProductInterface $product)
+    {
+        if (!$table = $this->getDocument()->find('css', 'table')) {
+            return false;
```

Comment:
But why? We are checking for the existence of a product. If there is no table, the page displays:
![zrzut ekranu 2016-03-11 o 10 34 47](https://cloud.githubusercontent.com/assets/13198869/13698545/f2926f5e-e774-11e5-9c50-8a71f79bf2c9.png)

which means that the `isThereProduct` method should return false. Am I right?